Orange/widgets/visualize/owsieve.py

```diff
@@ -200,9 +196,39 @@ def set_attr(self, attr_x, attr_y):
     def update_attr(self):
         """Update the graph and selection."""
         self.selection = set()
+        self.discrete_data = self.sparse_to_dense(self.data)
         self.update_graph()
         self.update_selection()
 
+    def sparse_to_dense(self, data, init=False):
+        """
+        Extracts two selected columns from sparse matrix.
+        GH-2260
+        """
+        def discretizer(data):
+            if any(attr.is_continuous for attr in chain(data.domain, data.domain.metas)):
+                discretize = Discretize(
+                    method=EqualFreq(n=4), remove_const=False,
+                    discretize_classes=True, discretize_metas=True)
+                return discretize(data)
+            return data
+
+        if not data.is_sparse() and init:
```

Comment:
Codeclimate is complaining about complexity. Will this code also work and hopefully reduce complexity?

```
if not data.is_sparse() and not init:
    return self.discrete_data
if data.is_sparse():
    attrs = {self.attr_x,
             self.attr_y}
    keys = []
    for attr in chain(self.data.domain, data.domain.metas):
        if attr in attrs:
            keys.append(attr.name)
    new_domain = data.domain.select_columns(keys)
    data = Table.from_table(new_domain, data)
    data.X = data.X.toarray()
return discretizer(data)
```